apps/app/src/app/shared/feature/the-favorites-modal/the-favorites-modal.component.ts

```diff
@@ -19,19 +24,41 @@ import { FavoritesService } from '../../util';
     TranslateModule,
     NgIf,
     AsyncPipe,
+    JsonPipe,
   ],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class TheFavoritesModalComponent {
+export class TheFavoritesModalComponent implements OnInit, OnDestroy {
+  private _eventsSubscription: Subscription | null = null;
+
   private _modalFavoriteOpened = new BehaviorSubject<boolean>(false);
 
+  public currentFavoriteName: string | null = null;
   public favoriteName = '';
 
   constructor(
     private _router: Router,
     private _favoritesService: FavoritesService
   ) {}
 
+  ngOnInit(): void {
+    this._eventsSubscription = this._router.events
+      .pipe(
+        filter(
+          (event): event is NavigationEnd => event instanceof NavigationEnd
```

Comment:
Je ne comprends pas à quoi sert le `event is NavigationEnd` ?
En tout cas comment s'utilise le `is` ici.